tests/test_torch_agent.py

```diff
@@ -32,34 +32,53 @@ class TestTorchAgent(unittest.TestCase):
     """Basic tests on the util functions in TorchAgent."""
 
     def test_vectorize(self):
+        """
+        Goal of this test is to make sure that the vectorize function is
+        actually adding a new field.
+        """
         try:
             from parlai.core.torch_agent import TorchAgent
         except ModuleNotFoundError as e:
             if 'pytorch' in e.msg:
                 print('Skipping TestTorchAgent.test_vectorize, no pytorch.')
                 return
-
         opt = {}
         opt['no_cuda'] = True
         opt['history_tokens'] = 10000
         opt['history_dialog'] = 10
         opt['history_replies'] = 'label_else_model'
-        dict = MockDict()
+        mdict = MockDict()
 
-        shared = {'opt': opt, 'dict': dict}
+        shared = {'opt': opt, 'dict': mdict}
         agent = TorchAgent(opt, shared)
         observation = {}
         observation["text"] = "What does the dog do?"
         observation["labels"] = ["The dog jumps over the cat."]
 
-        obs_vec = agent.vectorize(observation)
+        obs_vec = agent.vectorize(observation, addStartIdx=True,
+                                  addEndIdx=True)
         self.assertTrue('text_vec' in obs_vec,
                         "Field \'text_vec\' missing from vectorized observation")
         self.assertTrue(obs_vec['text_vec'].numpy().tolist() == [1, 3, 5],
                         "Vectorized text is incorrect.")
         self.assertTrue('labels_vec' in obs_vec,
                         "Field \'labels_vec\' missing from vectorized observation")
-        self.assertTrue(obs_vec['labels_vec'][0].numpy().tolist() == [1, 3, 5, dict.END_IDX],
+        self.assertTrue(obs_vec['labels_vec'][0].numpy().tolist() ==
+                        [mdict.START_IDX, 1, 3, 5, mdict.END_IDX],
```

Comment:
So this is a bit confusing because mdict.START_IDX == 1, so this response is currently [START, START, Word1, Word2, Word3, END].

I realize this is expected behavior given MockDict's definition of txt2vec. Perhaps it's just best this way.

Reply:
I see how that could be confusing to debug if you start failing the test. I'll just update MockDict to return numbers not reserved by the special tokens.